flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/api/utils/CustomStateBackendFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.api.utils;
+
+import org.apache.flink.configuration.IllegalConfigurationException;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.runtime.state.StateBackend;
+import org.apache.flink.runtime.state.StateBackendFactory;
+
+import java.io.IOException;
+
+/** A simple custom {@link StateBackendFactory} that throws an exception. */
+public class CustomStateBackendFactory implements StateBackendFactory<StateBackend> {
+
+    @Override
+    public StateBackend createFromConfig(ReadableConfig config, ClassLoader classLoader)
+            throws IllegalConfigurationException, IOException {
+        throw new ExpectedException();
```

Comment:
Do I understand correctly the purpose of this factory: verify that if a backend is not provided then it's loaded from config - by checking that `createFromConfig` is called?

If so, wouldn't it be possible to use `MockStateBackend` and then check maybe in operator that it is the actual backend? I think such check would be more reliable.

OTH, the effort might not worth it, so I'm also fine with the current approach.

Reply:
Yes, that is the intent. I went this route specifically because it was relatively straightforward to implement. I'd rather not change it if it is alright with you.